src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
and ``getDecoder`` documents it to let ``supports`` know about it.

But as far as ``decode`` is concerned, this exception is just an implementation detail (a decoder is free to do whatever it wants if you ask it to decode something for which it told you it does not support it). So IMO, documenting this in ``decode`` does not make sense (someone calling ``decode`` does not care about the error case ``If no decoder is found``, as it probably does not even care that it is calling a ChainDecoder rather than any DecoderInterface implementation, or at least it should not)